src/times.ts

```diff
@@ -1,14 +1,40 @@
 import { purry } from "./purry";
 
+type TimesArray<
+  T,
+  N extends number,
+  Iteration extends ReadonlyArray<unknown> = [],
+> = number extends N
+  ? // N is not a literal number, we can't deduce the type
+    Array<T>
+  : `${N}` extends `-${number}`
+    ? // N is non-positive, the mapper will never run
+      []
+    : `${N}` extends `${infer K extends number}.${number}`
+      ? // N is not an integer, we "floor" the number.
+        TimesArray<T, K, Iteration>
+      : N extends Iteration["length"]
+        ? // We finished building the output tuple
+          []
+        : // Add another item to the tuple and recurse.
+          [T, ...TimesArray<T, N, [unknown, ...Iteration]>];
+
 /**
  * Calls an input function `n` times, returning an array containing the results
  * of those function calls.
  *
  * `fn` is passed one argument: The current value of `n`, which begins at `0`
  * and is gradually incremented to `n - 1`.
  *
- * @param count - A value between `0` and `n - 1`. Increments after each function call.
- * @param fn - The function to invoke. Passed one argument, the current value of `n`.
+ * NOTE: The function returns a strict tuple with exactly N elements in it, but
+ * for very large literals TypeScript might not be able to compute it, for those
+ * cases cast the number to a primitive `number` (e.g. `1000 as number`) to
+ * bypass the refined typing.
+ *
```

Comment:
**suggestion:** we can prevent this on the type-level with `type-fest`'s [`GreaterThan`](https://github.com/sindresorhus/type-fest/blob/main/source/greater-than.d.ts). then we can remove this note?